core/resource/service.go

```diff
@@ -383,3 +400,85 @@ func (s Service) BulkCheckAuthz(ctx context.Context, resources []Resource, actio
 	}
 	return s.relationService.BulkCheckPermission(ctx, relations, actions)
 }
+
+func (s Service) ListUserResourcesByType(ctx context.Context, userID string, resourceType string) (ResourcePermission, error) {
+	user, err := s.userService.Get(ctx, userID)
+	if err != nil {
+		return ResourcePermission{}, err
+	}
+
+	res, err := s.listUserResources(ctx, resourceType, user)
+	if err != nil {
+		return ResourcePermission{}, err
+	}
+
+	return res, nil
+}
+
+func (s Service) ListAllUserResources(ctx context.Context, userID string, resourceType []string) (map[string]ResourcePermission, error) {
+	user, err := s.userService.Get(ctx, userID)
+	if err != nil {
+		return map[string]ResourcePermission{}, err
+	}
+
+	if len(resourceType) == 0 {
+		namespaces, err := s.namespaceService.List(ctx)
+		if err != nil {
+			return map[string]ResourcePermission{}, err
+		}
+
+		for _, ns := range namespaces {
+			if namespace.IsSystemNamespaceID(ns.ID) {
+				continue
+			}
+			resourceType = append(resourceType, ns.ID)
+		}
+	}
+
+	result := make(map[string]ResourcePermission)
+	for _, res := range resourceType {
+		if _, ok := result[res]; !ok {
+			list, err := s.listUserResources(ctx, res, user)
+			if err != nil {
+				return map[string]ResourcePermission{}, err
+			}
+			if len(list) != 0 {
+				result[res] = list
+			}
+		}
+	}
+
+	return result, nil
+}
+
+func (s Service) listUserResources(ctx context.Context, resourceType string, user user.User) (ResourcePermission, error) {
+	policies, err := s.policyService.List(ctx, policy.Filters{NamespaceID: resourceType})
+	if err != nil {
+		return ResourcePermission{}, err
+	}
+
+	res := make(ResourcePermission)
```

Comment:
lets make the variable clearer
i believe something like `resPermissionsMap` or `resPermissionsSet`
or better `Map` I would say, because `Set` probably similar with `Set` and `Get` if it is not in the context of Math or some Combinatorial term